async-download-rest/src/main/java/org/uniprot/api/async/download/config/AsyncRdfStreamerConfig.java

```diff
@@ -0,0 +1,67 @@
+package org.uniprot.api.async.download.config;
+
+import java.util.Collections;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.client.BufferingClientHttpRequestFactory;
+import org.springframework.http.client.ClientHttpRequestFactory;
+import org.springframework.http.client.SimpleClientHttpRequestFactory;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.DefaultUriBuilderFactory;
+import org.uniprot.api.common.repository.stream.rdf.*;
+import org.uniprot.api.rest.output.RequestResponseLoggingInterceptor;
+import org.uniprot.api.rest.service.TagPositionProvider;
+
+@Configuration
+public class AsyncRdfStreamerConfig {
+    private final PrologProvider prologProvider;
+    private final TagPositionProvider tagPositionProvider;
+    private final RdfEntryCountProvider rdfEntryCountProvider;
+
+    public AsyncRdfStreamerConfig(
+            PrologProvider prologProvider,
+            TagPositionProvider tagPositionProvider,
+            RdfEntryCountProvider rdfEntryCountProvider) {
+        this.prologProvider = prologProvider;
+        this.tagPositionProvider = tagPositionProvider;
+        this.rdfEntryCountProvider = rdfEntryCountProvider;
+    }
+
+    @Bean
+    public RdfStreamer asyncRdfStreamer(
```

Comment:
you ll probably be able to remove these configs by using different names uniprot, uniref etc